core/rpc/src/chain/number.rs

```diff
@@ -34,29 +35,22 @@ pub enum NumberOrHex<Number> {
 	Hex(U256),
 }
 
-impl<Number: traits::As<u64>> NumberOrHex<Number> {
+impl<Number: TryFrom<u64>> NumberOrHex<Number> {
 	/// Attempts to convert into concrete block number.
 	///
 	/// Fails in case hex number is too big.
 	pub fn to_number(self) -> Result<Number, String> {
-		let num: u64 = match self {
-			NumberOrHex::Number(n) => n.as_(),
+		match self {
+			NumberOrHex::Number(n) => Ok(n),
 			NumberOrHex::Hex(h) => {
-				// FIXME #1377 this only supports `u64` since `BlockNumber`
-				// is `As<u64>` we could possibly go with `u128`.
 				let l = h.low_u64();
 				if U256::from(l) != h {
-					return Err(format!("`{}` does not fit into the block number type.", h));
+					Err(format!("`{}` does not fit into u64 type; unsupported for now.", h))
 				} else {
-					l
+					Number::try_from(l)
+						.map_err(|_| format!("`{}` does not fit into block number type.", h))
 				}
 			},
-		};
-		// FIXME <2329>: Database seems to limit the block number to u32 for no reason
```

Comment:
I think this problem still exists.

Reply:
there's already an issue for it to be fixed. is this strictly needed?